src/Client.ts

```diff
@@ -52,29 +51,81 @@ export default class Client extends EventEmitter {
 	 * @param host - hostname or IP address of daemon
 	 * @param port - port number of daemon on the host
 	 */
-	introduce(host: string, port: number): void {
-		const client = connect(`http://${hostAndPort(host, port)}`)
+	public introduce(host: string, port: number): void {
+		const client = connect(`http://${Client.hostAndPort(host, port)}`)
 		client.on('error', (err) => this.emit('error', err))
-		this.available(client)
+		this.setAvailable(client)
 	}
 
-	private available(client: ClientHttp2Session): void {
-		const job = this.queue.pop()
-		if (!job) {
-			client.close(() => this.emit('done'))
-			return
+	/**
+	 * Mark a daemon as available.
+	 *
+	 * @param daemon - The daemon that is available.
+	 */
+	private setAvailable(daemon: ClientHttp2Session): void {
+		this.availableDaemons.add(daemon)
+		this.checkJobsAndAssign()
+	}
+
+	/**
+	 * Tries to assign the next available job to a daemon.
+	 * Closes all daemons if all jobs are finished.
+	 */
+	private checkJobsAndAssign(): void {
+		const job = this.jobOrderer.popNextJob()
+
+		if (job && this.availableDaemons.size > 0) {
+			const daemon = Array.from(this.availableDaemons.values())[0]
+			if (daemon) {
+				this.assignJobToDaemon(job, daemon)
+			} else {
+				throw 'skjdfhsadkfl'
```

Comment:
This condition is impossible, so don't check for it. It makes the code harder to read.

Also, consider allowing non-null assertions in the linter. That's a good concise way to indicate that something is always defined.

Reply:
lmao this might be the worst error message I've written